ast/src/lang/graphs/btreemap_graph.rs

```diff
@@ -699,41 +722,43 @@ impl Graph for BTreeMapGraph {
     fn count_edges_of_type(&self, edge_type: EdgeType) -> usize {
         self.edges
             .iter()
-            .filter(|(_, edge)| match (edge, &edge_type) {
+            .filter(|(_, _, edge)| match (edge, &edge_type) {
                 (EdgeType::Calls(_), EdgeType::Calls(_)) => true,
-                _ => **edge == edge_type,
+                _ => *edge == edge_type,
             })
             .count()
     }
 }
 
 impl BTreeMapGraph {
     fn find_node_from_node_ref(&self, node_ref: NodeRef) -> Option<Node> {
-        let prefix = format!(
-            "{:?}-{}-{}",
-            &node_ref.node_type, &node_ref.node_data.name, &node_ref.node_data.file
-        );
+        let prefix = format!("{:?}-", &node_ref.node_type).to_lowercase();
         self.nodes
             .range(prefix.clone()..)
             .take_while(|(k, _)| k.starts_with(&prefix))
+            .find(|(_, node)| {
+                node.node_data.file == node_ref.node_data.file
+                    && node.node_data.name == node_ref.node_data.name
+            })
             .map(|(_, node)| node.clone())
-            .next()
     }
 
     fn find_node_from_node_key(&self, node_type: NodeType, node_key: NodeKeys) -> Option<Node> {
```

Comment:
why does this not work with the ({}-{}-{}, type, name, file)? that will be so much more efficient